File: python/LeetCode/00762.py
from math import ceil, sqrt

def is_prime(self, n):
    if n < 2:
        return False

    sieve = [True] * (n + 1)
    sieve[0] = sieve[1] = False

    for i in range(2, ceil(sqrt(n)) + 1):
        if sieve[i]:
            for j in range(i * i, n + 1, i):
                sieve[j] = False

    return sieve[n]

def countPrimeSetBits(self, left: int, right: int) -> int:
    count = 0

    for i in range(left, right + 1):
        if self.is_prime(bin(i).count("1")):
            count += 1
    return count

File: go/leetcode/00762.go
package main

/*
Bounded by Right <= 10^6
10_000_000 = 0b100110001001011010000000
24 bits meaning 24 is max number of bits
*/

import (
	"math/bits"
)

var primes = [10]int{2, 3, 5, 7, 11, 13, 17, 19, 23, 29}

func countPrimeSetBits(left int, right int) int {
	count := 0
	for i := left; i <= right; i++ {
		b := bits.OnesCount(uint(i))
		for _, p := range primes {
			if b == p {
				count++
				break
			}
		}
	}
	return count
}
